src/main/java/uk/gov/hmcts/reform/sscs/service/CcdNotificationsPdfService.java

```diff
@@ -79,6 +81,52 @@ public SscsCaseData mergeCorrespondenceIntoCcd(SscsCaseData sscsCaseData, Corres
         return caseDetails.getData();
     }
 
+    /**
+     * This method generates PDF using HTML, stores in doc store and updates SSCS case data retrieved from DB with correspondence
+     *
+     * @param caseId - CCD case id
+     * @param correspondence - Correspondence which needs to be added to the case data
+     */
+    public void mergeCorrespondenceIntoCcdV2(Long caseId, Correspondence correspondence) {
+        Map<String, Object> placeholders = new HashMap<>();
+        placeholders.put("body", correspondence.getValue().getBody());
+        placeholders.put("subject", correspondence.getValue().getSubject());
+        placeholders.put("sentOn", correspondence.getValue().getSentOn());
+        placeholders.put("from", correspondence.getValue().getFrom());
+        placeholders.put("to", correspondence.getValue().getTo());
+
+        byte[] template;
+        try {
+            template = getSentEmailTemplate();
+        } catch (IOException e) {
+            throw new PdfGenerationException("Error getting template", e);
+        }
+
+        byte[] pdf = pdfServiceClient.generateFromHtml(template, placeholders);
+        String filename = String.format("%s %s.pdf", correspondence.getValue().getEventType(), correspondence.getValue().getSentOn());
+        List<SscsDocument> pdfDocuments = pdfStoreService.store(pdf, filename, correspondence.getValue().getCorrespondenceType().name());
+        final List<Correspondence> correspondences = pdfDocuments.stream().map(doc ->
+                correspondence.toBuilder().value(correspondence.getValue().toBuilder()
+                        .documentLink(doc.getValue().getDocumentLink())
+                        .build()).build()
+        ).toList();
```

Comment:
@nitinprabhuhmcts can this be moved into a function as it is duplicated from the original method and would reduce code duplication